x-pack/plugins/security_solution/common/api/timeline/model/components.gen.ts

```diff
@@ -0,0 +1,350 @@
+  type?: DataProviderType;
+}
+export interface DataProviderResultInput {
+  id?: string;
+  name?: string;
+  enabled?: boolean;
+  excluded?: boolean;
+  kqlQuery?: string;
+  queryMatch?: QueryMatchResult;
+  and?: DataProviderResultInput[];
+  type?: DataProviderType;
+}
+export const DataProviderResult: z.ZodType<
+  DataProviderResult,
+  ZodTypeDef,
+  DataProviderResultInput
+> = z.object({
+  id: z.string().optional(),
+  name: z.string().optional(),
+  enabled: z.boolean().optional(),
+  excluded: z.boolean().optional(),
+  kqlQuery: z.string().optional(),
+  queryMatch: QueryMatchResult.optional(),
+  and: z.array(z.lazy(() => DataProviderResult)).optional(),
+  type: DataProviderType.optional(),
+});
+
+export type RowRendererId = z.infer<typeof RowRendererId>;
+export const RowRendererId = z.enum([
+  'alert',
+  'alerts',
+  'auditd',
+  'auditd_file',
+  'library',
+  'netflow',
+  'plain',
+  'registry',
+  'suricata',
+  'system',
+  'system_dns',
+  'system_endgame_process',
+  'system_file',
+  'system_fim',
+  'system_security_event',
+  'system_socket',
+  'threat_match',
+  'zeek',
+]);
+export type RowRendererIdEnum = typeof RowRendererId.enum;
+export const RowRendererIdEnum = RowRendererId.enum;
+
+export type FavoriteTimelineResult = z.infer<typeof FavoriteTimelineResult>;
+export const FavoriteTimelineResult = z.object({
+  fullName: z.string().optional(),
+  userName: z.string().optional(),
+  favoriteDate: z.number().optional(),
+});
+
+export type FilterTimelineResult = z.infer<typeof FilterTimelineResult>;
+export const FilterTimelineResult = z.object({
+  exists: z.boolean().optional(),
+  meta: z
+    .object({
+      alias: z.string().optional(),
+      controlledBy: z.string().optional(),
+      disabled: z.boolean().optional(),
+      field: z.string().optional(),
+      formattedValue: z.string().optional(),
+      index: z.string().optional(),
+      key: z.string().optional(),
+      negate: z.boolean().optional(),
+      params: z.string().optional(),
+      type: z.string().optional(),
+      value: z.string().optional(),
+    })
+    .optional(),
+  match_all: z.string().optional(),
+  missing: z.string().optional(),
+  query: z.string().optional(),
+  range: z.string().optional(),
+  script: z.string().optional(),
+});
+
+export type SerializedFilterQueryResult = z.infer<typeof SerializedFilterQueryResult>;
+export const SerializedFilterQueryResult = z.object({
+  filterQuery: z
+    .object({
+      kuery: z
+        .object({
+          kind: z.string().optional(),
+          expression: z.string().optional(),
+        })
+        .optional(),
+      serializedQuery: z.string().optional(),
+    })
+    .optional(),
+});
+
+export type Sort = z.infer<typeof Sort>;
+export const Sort = z.object({
+  columnId: z.string().optional(),
+  sortDirection: z.string().optional(),
+});
+
+export type SavedTimeline = z.infer<typeof SavedTimeline>;
+export const SavedTimeline = z.object({
```

Comment:
A lot of these fields are `nullable` based on the existing io-ts type: https://github.com/elastic/kibana/blob/c325e3caf2eca378dee077dd421bd0defeb1a48a/x-pack/plugins/security_solution/common/api/timeline/model/api.ts#L344

Reply:
[Fixed here](https://github.com/elastic/kibana/pull/188844/commits/f70299c8c1e32d4b0d22ccb3580ae918520db71a)